### PR TITLE
Add crud products

### PR DESCRIPTION
**Pull request by mo1502**

### CODE REVIEW

File: app/controllers/api/v1/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            render json: { data: ActiveModel::SerializableResource.new(Product.all, each_serializer: ProductSerializer) }
+        end
+
+        def show
+            render json: ProductSerializer.new(@product).to_h
+          end
+
+
+        def create
+        @product = Product.new(product_params)
```
Comment (PatrykTyszka) [inline app/controllers/api/v1/products_controller.rb:16]: fix indetnation

File: app/controllers/api/v1/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            else
+              render json: { errors: @product.errors.messages }, status: :unprocessable_entity
+            end
+          end
+        
+          def destroy
+            if @product.destroy
+              render json: @product
+            else
+              render json: { errors: @product.errors.messages }, status: :unprocessable_entity
```
Comment (PatrykTyszka) [inline app/controllers/api/v1/products_controller.rb:36]: never happened coz you don't have validation on destroy.

File: config/routes.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   namespace :api do
     namespace :v1 do
+      resources :products
```
Comment (h4b00) [inline config/routes.rb:5]: only: [:index, :destroy, :show, :update, :create]

File: app/serializers/product_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class ProductSerializer < ActiveModel::Serializer
+    attributes :id, :name, :quantity
+  end
```
Comment (h4b00) [inline app/serializers/product_serializer.rb:3]: indentation

File: app/models/products_recipes.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class ProductRecipe < ApplicationRecord
+    belongs_to :product
+    belongs_to :recipe
+  end
```
Comment (h4b00) [inline app/models/products_recipes.rb:4]: indentation

File: app/models/product.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class Product < ApplicationRecord
+  include ActiveModel::Validations
```
Comment (h4b00) [inline app/models/product.rb:2]: we don't need this

File: db/migrate/20210819070931_add_quantity_to_product.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddQuantityToProduct < ActiveRecord::Migration[6.1]
+  def change
+    add_column :products, :quantity, :integer
```
Comment (h4b00) [inline db/migrate/20210819070931_add_quantity_to_product.rb:3]: `null: false, default: 1`

File: app/controllers/api/v1/products_controller.rb [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+          params.require(:product).permit(%i[name quantity])
+        end
+
+        def set_product
+            @product = Product.find(params[:id])
+          end
+      end
+    end
+  end
+
```
Comment (h4b00) [inline app/controllers/api/v1/products_controller.rb:53]: fix indentations.

Comment (h4b00) [inline app/controllers/api/v1/products_controller.rb:53]: remove empty line at the end

**Review by h4b00 — CHANGES_REQUESTED** [review]

**Review by h4b00 — APPROVED** [review]

**Pull request merged**